cli/cmd/main.go

```diff
@@ -712,6 +793,70 @@ func DataCommand(c *cli.Context) error {
 	}
 
 	filter := &datapb.Filter{}
+	if err := createDataFilter(c, filter); err != nil {
+		return err
+	}
+
+	client, err := rdkcli.NewAppClient(c)
+	if err != nil {
+		return err
+	}
+
+	dataType := c.String(dataFlagDataType)
+	switch dataType {
+	case dataTypeBinary:
+		if err := client.BinaryData(c.String(dataFlagDestination), filter, c.Uint(dataFlagParallelDownloads)); err != nil {
+			return err
+		}
+	case dataTypeTabular:
+		if err := client.TabularData(c.String(dataFlagDestination), filter); err != nil {
+			return err
+		}
+	default:
+		return errors.Errorf("invalid data type %s", dataType)
+	}
+
+	return nil
+}
+
+// DeleteCommand runs the command for deleting data from the Viam cloud.
+func DeleteCommand(c *cli.Context) error {
+	if c.String(dataFlagDataType) != dataTypeBinary && c.String(dataFlagDataType) != dataTypeTabular {
+		return errors.Errorf("type must be binary or tabular, got %s", c.String("type"))
+	}
+
+	filter := &datapb.Filter{}
+	if err := createDataFilter(c, filter); err != nil {
+		return err
+	}
+
+	client, err := rdkcli.NewAppClient(c)
+	if err != nil {
+		return err
+	}
+
+	dataType := c.String(dataFlagDataType)
+	switch dataType {
+	case dataTypeBinary:
+		if err := client.DeleteBinaryData(filter); err != nil {
+			return err
+		}
+	case dataTypeTabular:
+		if err := client.DeleteTabularData(filter); err != nil {
+			return err
+		}
+	default:
+		return errors.Errorf("invalid data type %s", dataType)
+	}
+
+	return nil
+}
+
+func createDataFilter(c *cli.Context, filter *datapb.Filter) error {
+	if c.String(dataFlagDataType) != dataTypeBinary && c.String(dataFlagDataType) != dataTypeTabular {
```

Comment:
I think this validation makes sense to include somewhere outside of this function, since it isn't directly filter related (in the sense that I think it might be appropriate to include if it was validating something needed to build the filter, but the `dataFlagDataType` isn't otherwise used in here)

Reply:
related to above